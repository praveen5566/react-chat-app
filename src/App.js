import React from 'react';
import './App.scss';
import { UserNameInput } from './components/userNameInput/UserNameInput';
import { ChatWindow } from './components/chatWindow/ChatWindow';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';

function App() {
  return (
    <Router>
      <Redirect exact from="/" to="/login" />
      <Route path="/login" component={UserNameInput} />
      <Route path="/chatWindow" component={ChatWindow} />
    </Router>
  );
}

export default App;
