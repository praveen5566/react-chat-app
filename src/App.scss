*,
*:before,
*:after {
  box-sizing: border-box;
}

:root {
  --app-color-white: white;
  --app-theme-color: rgb(0, 161, 235);
  --app-theme-color-hover: rgb(45, 65, 103);
  --app-color-grey: rgb(247, 247, 247);
  --app-dark-font-color: black;
  --app-light-font-color: rgb(117, 113, 113);
}

body {
  margin: 0;
  background-color: var(--app-color-white);
}

header {
  font-family: Arial Black;
}
