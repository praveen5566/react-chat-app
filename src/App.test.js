import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import { BrowserRouter as Router, MemoryRouter, Route, Redirect } from 'react-router-dom';

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('should check routes', () => {
  const wrapper = mount(<MemoryRouter initialEntries={['/']}>
    <App />
  </MemoryRouter>);
  expect(wrapper.find(Router).length).toEqual(1);
  expect(wrapper.find(Redirect).length).toEqual(1);
  expect(wrapper.find(Route).length).toEqual(2);
});
