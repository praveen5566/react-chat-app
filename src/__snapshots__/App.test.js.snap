// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BrowserRouter>
  <Redirect
    exact={true}
    from="/"
    to="/login"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/chatWindow"
  />
</BrowserRouter>
`;
